compiler/ast/src/passes/reconstructor.rs

```diff
@@ -172,10 +172,7 @@ pub trait ExpressionReconstructor {
                     .into_iter()
                     .map(|member| StructVariableInitializer {
                         identifier: member.identifier,
-                        expression: match member.expression {
-                            Some(expression) => Some(self.reconstruct_expression(expression).0),
-                            None => Some(self.reconstruct_expression(Expression::Identifier(member.identifier)).0),
-                        },
+                        expression: member.expression.map(|expr| self.reconstruct_expression(expr).0),
```

Comment:
In the `None` case, the `identifier` is the expression. Would the new code skip that case? Maybe it makes sense to reorganize the `StructVariableInitializer` so that `identifier` is optional?